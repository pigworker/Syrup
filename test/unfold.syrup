display nand

not(<Bit>) -> <Bit>
not(X) = nand(X,X)

display [nand] not

f(<Bit>) -> <Bit>
f(X) = Z

display f

display [g] not