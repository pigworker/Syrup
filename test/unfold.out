Warning: I don't have an implementation for nand.

Circuit not is defined.

Displaying not (with nand unfolded):
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
 "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Generated by graphviz version 2.43.0 (0)
 -->
<!-- Title: whitebox Pages: 1 -->
<svg width="126pt" height="334pt"
 viewBox="0.00 0.00 126.00 334.00" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 330)">
<title>whitebox</title>
<polygon fill="white" stroke="transparent" points="-4,4 -4,-330 122,-330 122,4 -4,4"/>
<g id="clust1" class="cluster">
<title>cluster_circuit__GATE_not_3</title>
</g>
<!-- GATE_not_3__INPUTS -->
<g id="node1" class="node">
<title>GATE_not_3__INPUTS</title>
<text text-anchor="start" x="52" y="-294" font-family="Times,serif" font-size="20.00">X</text>
</g>
<!-- COPY_3__1 -->
<g id="node3" class="node">
<title>COPY_3__1</title>
<polygon fill="skyblue" stroke="transparent" points="65,-221 55,-221 55,-214 65,-214 65,-221"/>
</g>
<!-- GATE_not_3__INPUTS&#45;&gt;COPY_3__1 -->
<g id="edge4" class="edge">
<title>GATE_not_3__INPUTS:X&#45;&gt;COPY_3__1:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M60,-285.97C60,-271.07 60,-244.91 60,-222.5"/>
<text text-anchor="middle" x="62.5" y="-242.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- GATE_nand_3__0__0 -->
<g id="node2" class="node">
<title>GATE_nand_3__0__0</title>
<text text-anchor="start" x="41" y="-146.4" font-family="Times,serif" font-size="7.00">X</text>
<text text-anchor="start" x="73" y="-146.4" font-family="Times,serif" font-size="7.00">Y</text>
<text text-anchor="start" x="34" y="-120" font-family="Times,serif" font-size="20.00">nand</text>
<text text-anchor="start" x="41.5" y="-100.4" font-family="Times,serif" font-size="7.00">Z</text>
<polygon fill="none" stroke="black" points="24.5,-91 24.5,-159 93.5,-159 93.5,-91 24.5,-91"/>
</g>
<!-- GATE_not_3__OUTPUTS -->
<g id="node4" class="node">
<title>GATE_not_3__OUTPUTS</title>
</g>
<!-- GATE_nand_3__0__0&#45;&gt;GATE_not_3__OUTPUTS -->
<g id="edge3" class="edge">
<title>GATE_nand_3__0__0:s&#45;&gt;GATE_not_3__OUTPUTS:__VIRTUAL__2</title>
<path fill="none" stroke="black" stroke-width="2" d="M44,-95C44,-69.17 44,-38.41 44,-25.36"/>
<polygon fill="black" stroke="black" stroke-width="2" points="45.75,-25.07 44,-20.07 42.25,-25.07 45.75,-25.07"/>
<text text-anchor="middle" x="46.5" y="-57.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_3__1&#45;&gt;GATE_nand_3__0__0 -->
<g id="edge1" class="edge">
<title>COPY_3__1:s&#45;&gt;GATE_nand_3__0__0:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M58,-214C58,-205.57 54.15,-204.16 52,-196 47.76,-179.87 44.68,-174.61 44.1,-160.25"/>
<polygon fill="black" stroke="black" stroke-width="2" points="45.84,-159.97 44,-155 42.35,-160.03 45.84,-159.97"/>
<text text-anchor="middle" x="54.5" y="-184.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_3__1&#45;&gt;GATE_nand_3__0__0 -->
<g id="edge2" class="edge">
<title>COPY_3__1:s&#45;&gt;GATE_nand_3__0__0:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M62,-214C62,-188.84 74.2,-182.72 75.82,-160.09"/>
<polygon fill="black" stroke="black" stroke-width="2" points="77.58,-160.06 76,-155 74.08,-159.94 77.58,-160.06"/>
<text text-anchor="middle" x="72.5" y="-184.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
</g>
</svg>

Error: I was trying to make sense of the following code:

f(<Bit>) -> <Bit>
f(X) = Z


Error: You tried to use Z but it is not in scope.


Warning: Circuit f has been stubbed out.

Warning: I don't have an implementation for f.

Error: You tried to use g but it is not in scope.

